--- /moonraker/components/file_manager/file_manager.py.old	2023-12-05 13:01:48.905681044 +0800
+++ /moonraker/components/file_manager/file_manager.py	    2023-12-05 13:03:18.237681049 +0800
@@ -244,7 +244,8 @@ class FileManager:
         return new_path
 
     def disable_write_access(self):
-        self.full_access_roots.clear()
+        logging.info("File System disable_write_access function ignored for droidvm")
+        # self.full_access_roots.clear()
 
     def check_write_enabled(self):
         if not self.full_access_roots:

--- /moonraker/components/machine.py.old	2023-12-09 15:11:00.000000000 +0800
+++ /moonraker/components/machine.py	2023-12-09 15:14:09.686909700 +0800
@@ -462,6 +462,8 @@ class Machine:
         )
 
     def _get_sdcard_info(self) -> Dict[str, Any]:
+        # 在部分root过的机型上会去读取真实sdcard信息，导致moon启动不了，这里不让它读了
+        return {}
         sd_info: Dict[str, Any] = {}
         cid_file = pathlib.Path(SD_CID_PATH)
         if not cid_file.exists():
